internal/controller/config.go

```diff
@@ -40,7 +41,9 @@ type Config struct {
 	Reporter           reporter.Reporter
 	ExecutableReporter reporter.ExecutableReporter
 
-	Fs *flag.FlagSet
+	// If ReporterFactory is set, it will be used to create a Reporter and set it as the Reporter field.
+	ReporterFactory func(cfg *reporter.Config, nextConsumer xconsumer.Profiles) (reporter.Reporter, error)
```

Comment:
Nit, but shouldn't this option be grouped with ExecutableReporter instead?
Also, possibly for another PR, how about grouping the configs that are public due to how our packages are done, but shouldn't be used (`Reporter`) together with a specific comment?